### ISSUE TITLE
$adWordsServices->get doesn't work in local anymore

### ISSUE THREAD

**Issue by SeriousJ55** [googleads/googleads-php-lib#384]
I use v201705 of the API with the PHP Library.

It used to work fine both in local on my computer running wamp and on my Apache server.

I just realized that the simple function to get a service doesn't work anymore in local. The following line just disconnects me:

`$adWordsServices->get($session,` CampaignService::class);

I don't even get an error message. And it works fine on my Apache server. And I haven't changed anything in my code.

What's wrong?

**Comment by SeriousJ55** [comment]
When running getCampains.php in command line, it stops working and I get "CLI has stopped working" error from Windows

**Comment by fiboknacky** [comment]
Hello @SeriousJ55 

[Looks like](https://stackoverflow.com/questions/1671995/php-cli-has-stopped-working) it's a problem with your PHP, not related directly to the library.
Can you run other PHP programs?

Best,
Knack

**Comment by SeriousJ55** [comment]
Yes, actually. I just tried it on a different computer, with a different environment, and I get the exact same problem.

Please help.

**Comment by SeriousJ55** [comment]
And I already tried all the solutions in the Stackoverflow post. None of them worked. The problem seems to be very specific to your SDK.

**Comment by fiboknacky** [comment]
Could you specify how you tried the library? Installing it using composer from scratch and run GetCampaigns example right away?
Do you use Windows with XAMPP? What's your PHP version?

**Comment by SeriousJ55** [comment]
I installed the library from scratch using composer. I run GetCampaings.php from the command line.

I use WampServer Version 3.0.4 32bit installed on Windows 10. My PHP version is 7.0.4.

The library itself is properly installed. I can authenticate and get a proper session object. I get the error at the following line:

`$adWordsServices->get($session,` CampaignService::class);

Everything else works fine - and even the library used to work fine until recently.

**Comment by SeriousJ55** [comment]
It would be so nice to have it working without reinstalling all our dev environment. We already reinstalled 3 times because of the library since we started using it 2.5 years ago. It's the only library that doesn't work just out of the box. Heeeelp!

**Comment by SeriousJ55** [comment]
OK, I found a solution after two days working on it.

I had already put a generic permission file "cacert.pem" and declared it in php.ini by setting up the CURLOPT_CAINFO option, like this:
`curl.cainfo` ="C:\wamp\cacert.pem"

Now I had to also declare it by setting up the openssl.cafile to the same permission file, like this:
`openssl.cafile="C:\wamp\cacert.pem"`

Everything is explained here:
https://stackoverflow.com/questions/29822686/curl-error-60-ssl-certificate-unable-to-get-local-issuer-certificate

The first proposed solution worked the first time. I had to apply the second solution to make it work this time. Who knows why?

Now please allow me to vent against Google.

<venting>
We're getting fed up with Google API and PHP SDK. Nothing works the first time, ever. And the API and SDK keep changing, sometimes with no backward compatibility. So every couple of months, we have to loose several days, or weeks, trying to make everything work again, configuring obscure and random parameters until something finally works. We even had to reinstall our dev environment, three times!

We work with dozens of APIs and SDKs. Some are just amazingly easy to use (hello Stripe, hello PHPdocx), some are less straightforward (yes, you Paypal). But none of them is nearly as difficult to use as Google's API.

Anyone listening at Google?

</venting>

**Comment by fiboknacky** [comment]
Hello,

I'm sorry that the library doesn't work with your environment many times.
Making it work well with PHP under Windows is what we're improving.
The certification issues in Windows are persistent, elusive and quite various and this happens to many developers who rely on SSL connections, not limited to only this library, I think.
I'm trying to collecting common problems and will add them to `README.md` soon when ready.

> And the API and SDK keep changing, sometimes with no backward compatibility. So every couple of months, we have to loose several days, or weeks, trying to make everything work again, configuring obscure and random parameters until something finally works. 

Regarding this, I'd like to explain a bit more. We do have a large change in the library but only once, back in December last year when the main branch of this library is shifted to the current one with a large number of benefits, which you can take a look [here](https://github.com/googleads/googleads-php-lib/blob/master/UPGRADING.md#new-ads-php-client-library-upgrade-guide).

Apart from that, when we introduce a breaking change, we always follow the [semantic versioning](semver.org) to warn that the release can introduce some backward incompatibilities. In most other cases, we don't expect developers to reinstall the dev environment. Doing `composer update` should just load everything to your environment and that's it.

We do value your feedback so please feel free to give opinions and feedback as always. ;)
It looks like you could resolve your issue so I'll close this. Feel free to re-open if related issues re-occur.

Cheers,
Knack
